core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/describe.kt

```diff
@@ -117,12 +117,11 @@ private fun DataColumn<Any?>.convertToComparableOrNull(): DataColumn<Comparable<
         valuesAreComparable() -> asComparable()
 
         // Found incomparable number types, convert all to Double first
-        isNumber() -> cast<Number?>().map {
-            if (it?.isPrimitiveNumber() == false) {
-                // Cannot calculate statistics of a non-primitive number type
+        isNumber() -> with(cast<Number?>()) {
```

Comment:
I'd revert this back to `map` with non-local break. Then all values are visited just once, now they are visited twice.